### DISCUSSION TITLE
To be able to define a preset with some color pickers on

### DISCUSSION THREAD

**Discussion by mmoqui** [aurelienpierreeng/ansel#95] [Ideas]
It would be nice to be able to define a preset in a module with some of its color pickers enabled so that for each image on which the preset is applied the picker will also compute its associated task for the image.

For example, with this,  I could define a preset in Filmic RGB with some chosen options and with the auto-tune color picker on. By applying this preset, automatically Filmic will compute the values of the white and black relative exposures and from these I can then just adjust slightly the values per image. Idem in Color Balance RGB with the color picker for the white fulcrum in the masks table when applying some peculiar color/brightness adjustments in the shadow/middle/highlights.

**Comment by aurelienpierre** [comment]
What you think you want is a preset with color pickers.

What you actually want is white/black (max/min RGB) auto-detected and auto-set from what is in the pipeline, and that has nothing to do with color pickers and preset.

**Comment by mmoqui** [comment]
I don't know if what you wrote is what I want eventually. Or even whether this is desired or a good suggestion.

I think it would be better to explain from what the idea comes from.

For instance, when I work on a bunch of  RAWs, in order to avoid redundant tasks/settings, I usually create a preset in each module and these presets are specific to a given context (for a given (mini-)serie for example). From them, I can also create a style so I can apply a given development on some well-chosen images over the time. A preset in Filmic can be, for example, a color preservation mode to apply. But when I create a preset, all the settings in Filmic are registered, even the white and black relative exposures. But I don't want these settings to be registered into the preset because they are relative to the specific scene of the image. 
So, in my mind, as in my work I usually click on the color picker of the white relative exposure to compute the white relative exposure on the whole image and then fine-adjust the computed result with the slider, I thought it would be nice to have the activated state of the color picker (with the area on which it is applied) to be registered in the preset instead of the values.

Idem with the Color Balance RGB in which I click always, in the mask tab, the color picker of the white fulcrum when working on colors. This is to avoid to go back in each of the images on which a preset is applied in order to adjust the fulcrum.

The idea here is to have presets to be able to register not only values but also  "command" like for example: (for Color Balance RGB) apply now the computation of the white fulcrum or (for Filmic) the white relative exposure on the whole image.

But if this is difficult to code, perhaps a better solution or an intermediate solution is to avoid, in Filmic, when creating a preset, to register both the white and black relative exposures (or to register them only in the case they are modified by the user).
